pkg/kubernetes/kube.go

```diff
@@ -271,7 +275,7 @@ func (kc *Client) ResourceShouldConvergeToSelector(resourceFileName, selector st
 	gvr, resource := unstructuredResource.GVR, unstructuredResource.Resource
 
 	for {
-		if counter >= DefaultWaiterRetries {
+		if counter >= kc.getWaiterTries() {
```

Comment:
This needs to be > or you could move it to the end of the loop before the sleep and leave as >= so that it doesn't sleep on the last iteration before returning the error.

Reply:
If I use `>` and not `>=` its going to do `kc.getWaiterTries() + 1` because `counter` starts from `0`. 
What behavior you want to achieve? based on our conversation I see two options:

1. We call it `WaiterRetries` and implement it such that the total number of tries is `WaiterRetries + 1`. That way we can use `SetWaiterRetries` as `0`.
2. We call it `WaiterTries` and implement it such that the total number of tries is `WaiterTries`. That way `SetWaiterRetries` has to be minimum `1`.

The last commit I sent uses option 2. Do you want to do option 1? or do you have something else in mind?

Reply:
You are correct.  It will be 0 the first time and work.